examples/platform/silabs/efr32/rs911x/hal/efx_spi.c

```diff
@@ -163,115 +150,24 @@ void rsi_hal_board_init(void)
 }
 
 /*****************************************************************************
- *@fn static bool dma_complete_cb(unsigned int channel, unsigned int sequenceNo, void *userParam)
- *
  *@brief
- *    DMA transfer completion callback. Called by the DMA interrupt handler.
- *    This is being set in the tx/rx of the DMA
- *
- * @param[in] channel:
- * @param[in] sequenceNO: sequence number
- * @param[in] userParam :user parameter
+ *    Spi dma transfer is complete Callback
+ *    Notify the task that initiated the SPI transfer that it is completed.
+ *    The callback needs is a SPIDRV_Callback_t function pointer type
+ * @param[in] pxHandle: spidrv instance handle
+ * @param[in] transferStatus: Error code linked to the completed spi transfer. As master, the return code is irrelevant
+ * @param[in] lCount: number of bytes transferred.
  *
  * @return
  *    None
  ******************************************************************************/
-static bool dma_complete_cb(unsigned int channel, unsigned int sequenceNo, void * userParam)
+static void spi_dmaTransfertComplete(SPIDRV_HandleData_t * pxHandle, Ecode_t transferStatus, int itemsTransferred)
 {
+    configASSERT(spiInitiatorTaskHandle != NULL);
     BaseType_t xHigherPriorityTaskWoken = pdFALSE;
-    // uint8_t *buf = (void *)userParam;
-
-    (void) channel;
-    (void) sequenceNo;
-    (void) userParam;
-
-    // WFX_RSI_LOG ("SPI: DMA done [%x,%x,%x,%x]", buf [0], buf [1], buf [2], buf [3]);
-    xSemaphoreGiveFromISR(spi_sem, &xHigherPriorityTaskWoken);
+    vTaskNotifyGiveFromISR(spiInitiatorTaskHandle, &xHigherPriorityTaskWoken);
+    spiInitiatorTaskHandle = NULL;
     portYIELD_FROM_ISR(xHigherPriorityTaskWoken);
```

Comment:
so this is always called from an ISR context ?

Reply:
Yes it is a DMA completion callback the DMA completion is triggered by a IRQ